rsocket-core/src/test/java/io/rsocket/core/RSocketRequesterTest.java

```diff
@@ -333,6 +346,124 @@ public void shouldThrownExceptionIfGivenPayloadIsExitsSizeAllowanceWithNoFragmen
         .verify();
   }
 
+
+  private static Stream<Arguments> racingCases() {
+    return Stream.of(
+            Arguments.of(
+                    (Runnable) () -> System.out.println("RequestChannel downstream cancellation case"),
+                    (Function<ClientSocketRule, Publisher<Payload>>) (rule) -> rule.socket.requestChannel(Flux.just(EmptyPayload.INSTANCE)),
+                    (BiConsumer<AssertSubscriber<Payload>, ClientSocketRule>) (as, rule) -> {
+                      LeaksTrackingByteBufAllocator allocator = LeaksTrackingByteBufAllocator.instrumentDefault();
+                      ByteBuf metadata = allocator.buffer();
+                      metadata.writeCharSequence("abc", CharsetUtil.UTF_8);
+                      ByteBuf data = allocator.buffer();
+                      data.writeCharSequence("def", CharsetUtil.UTF_8);
+                      int streamId = rule.getStreamIdForRequestType(REQUEST_CHANNEL);
+                      ByteBuf frame = PayloadFrameFlyweight.encode(allocator, streamId, false, false, true, metadata, data);
+
+                      RaceTestUtils.race(as::cancel, () -> rule.connection.addToReceivedBuffer(frame));
+                    }
+            )/*,*/
```

Comment:
Should we instrument all tests in this class with the leak allocator, as well as `RSocketResponderTest` and others?

Reply:
Yes, working on the responder test but in any case, I ignore them since reactor is leaking so the test can not be passed because of that